ngmix/tests/test_ksigmamom.py

```diff
@@ -0,0 +1,317 @@
+        dvdx=gs_wcs.dvdx, dvdy=gs_wcs.dvdy)
+
+    gal = galsim.Gaussian(
+        fwhm=fwhm
+    ).shear(
+        g1=-0.1, g2=0.2
+    ).withFlux(
+        400
+    ).shift(
+        dx=shift[0], dy=shift[1]
+    )
+    psf = galsim.Gaussian(
+        fwhm=psf_fwhm
+    ).shear(
+        g1=0.3, g2=-0.15
+    )
+    im = galsim.Convolve([gal, psf]).drawImage(
+        nx=image_size,
+        ny=image_size,
+        wcs=gs_wcs
+    ).array
+    noise = np.sqrt(np.sum(im**2)) / snr
+    wgt = np.ones_like(im) / noise**2
+
+    psf_im = psf.shift(
+        dx=psf_shift[0], dy=psf_shift[1]
+    ).drawImage(
+        nx=image_size,
+        ny=image_size,
+        wcs=gs_wcs
+    ).array
+
+    g1arr = []
+    g2arr = []
+    Tarr = []
+    farr = []
+    momarr = []
+    snrarr = []
+    fitter = KSigmaMom(
+        fwhm=mom_fwhm,
+        pad_factor=pad_factor,
+    )
+
+    # get true flux
+    im_true = gal.drawImage(
+        nx=image_size,
+        ny=image_size,
+        wcs=gs_wcs,
+        method='no_pixel').array
+    obs = Observation(
+        image=im_true,
+        jacobian=jac,
+    )
+    res = KSigmaMom(fwhm=mom_fwhm).go(obs=obs, no_psf=True)
+    flux_true = res["flux"]
+    T_true = res["T"]
+    g1_true = res["e"][0]
+    g2_true = res["e"][1]
+
+    if snr > 1e4:
+        nitr = 10000
+    else:
+        nitr = 100
+    for _ in range(nitr):
+        _im = im + rng.normal(size=im.shape, scale=noise)
+        obs = Observation(
+            image=_im,
+            weight=wgt,
+            jacobian=jac,
+            psf=Observation(image=psf_im, jacobian=psf_jac),
+        )
+
+        res = fitter.go(obs=obs)
+        if res['flags'] == 0:
+            _g1, _g2 = res['e'][0], res['e'][1]
+            g1arr.append(_g1)
+            g2arr.append(_g2)
+            Tarr.append(res['T'])
+            farr.append(res['flux'])
+            snrarr.append(res["flux"] / res["flux_err"])
+            momarr.append(res["mom"])
+
+    print("\n")
+    _report_info("snr", snrarr, None, None)
+    _report_info("flux", farr, flux_true, res["flux_err"])
+    _report_info("T", Tarr, T_true, res["T_err"])
+    _report_info("g1", g1arr, g1_true, res["e_err"][0])
+    _report_info("g2", g2arr, g2_true, res["e_err"][1])
+    mom_cov = np.cov(np.array(momarr).T)
+    print("mom cov ratio:\n", res["mom_cov"]/mom_cov, flush=True)
+    assert np.allclose(np.mean(farr), flux_true, atol=0, rtol=0.1)
+    assert np.allclose(np.std(farr), res["flux_err"], atol=0, rtol=0.2)
+
+
+@pytest.mark.parametrize('snr', [1e2])
+@pytest.mark.parametrize('pixel_scale', [0.25])
+@pytest.mark.parametrize('fwhm,psf_fwhm', [(2.0, 1.0)])
+@pytest.mark.parametrize('mom_fwhm', [2.0])
+@pytest.mark.parametrize('image_size', [58])
+@pytest.mark.parametrize('pad_factor', [1.5])
+def test_ksigmamom_mn_cov(
+    pad_factor, image_size, fwhm, psf_fwhm, pixel_scale, snr, mom_fwhm,
+):
+    """Slower test to make sure means and errors are right in a well-sampled case."""
```

Comment:
does this mean that the pixelization is the major reason values and errors are incorrect?

Reply:
"Well-sampled" refers to running a bunch of monte carlo realizations, not the pixels. I will clarify.